### DISCUSSION TITLE
G-CLI with GitLab Runner timeouts most of the time

### DISCUSSION THREAD

**Discussion by Benni0706** [JamesMc86/G-CLI#145] [Q&A]
I use G-CLI with GitLab pipelines. Unfortunately, I am encountering an error that seems to occur randomly and without any pattern. In most cases I get the following error:

Error: No connection established with application.
Caused by:
    Timed out waiting for app to connect to g-cli
Location:
    src\main.rs:62:10

However, the error does not always occur, in other cases everything works fine. When I run the g-cli command I use directly locally, there are no problems. I have also checked that when I start LabVIEW, no dialogues are displayed and my code is not broken.

I am not sure if this error is due to G-CLI at all, but since I am currently at a dead end, I wanted to ask anyway.

**Comment by JamesWiresmith** [comment]
Hi Benni,

Yep these are hard ones to troubleshoot. The next step would be to run G-CLI in verbose mode (-v flag). That will give us a little more detail about what it has tried and may offer some clues. Though it is hard for it to detect the exact state of LabVIEW - we may at least see if it picks up the LabVIEW process existing.

**Comment by Benni0706** [comment]
Hi James,

The G-CLI command itself seems to be executed and the LabVIEW process is launched. After that, however, nothing happens until the error message appears. On the computer on which the GitLab Runner is running, I can also see that the LabVIEW process is started but never opens as a window.

```
Executing "step_script" stage of the job script 00:49
$ g-cli --timeout 30000 -v --lv-ver 2022 execute_analyzer.vi -- "results.txt" "$CI_PROJECT_DIR\VI_Analyzer_Config.viancfg"
06:24:09.664 [DEBUG] G CLI Started - Verbose Mode
06:24:09.664 [DEBUG] Version 3.0.0
06:24:09.664 [DEBUG] G CLI Arguments: "C:\Program Files\G-CLI\bin\g-cli.exe" "--timeout" "30000" "-v" "--lv-ver" "2022" "execute_analyzer.vi" "--" "results.txt" "C:\GitLab-Runner\builds\NyzQ9DM_\0\it-ed-labview\msel-ci-ex-simple\VI_Analyzer_Config.viancfg"
06:24:09.664 [DEBUG] Arguments passed to LabVIEW: "results.txt C:\\GitLab-Runner\\builds\\NyzQ9DM_\\0\\it-ed-labview\\msel-ci-ex-simple\\VI_Analyzer_Config.viancfg"
06:24:09.667 [DEBUG] Detected LabVIEW versions:
2022 Q3, 32bit (C:\Program Files (x86)\National Instruments\LabVIEW 2022\)
06:24:09.667 [DEBUG] Looks like VI "execute_analyzer.vi" doesn't exist - Checking in vi.lib/G CLI Tools instead.
06:24:11.715 [DEBUG] Launching: C:\Program Files (x86)\National Instruments\LabVIEW 2022\LabVIEW.exe "C:\\Program Files (x86)\\National Instruments\\LabVIEW 2022\\vi.lib\\G CLI Tools\\execute_analyzer.vi -unattended -- -p:5000"
06:24:11.726 [DEBUG] Process launched with PID 7116
Error: No connection established with application.
Caused by:
    Timed out waiting for app to connect to g-cli
Location:
    src\main.rs:62:10
```

**Comment by JamesWiresmith** [comment]
Could you have two instances running?

There was a bug in 3.0.0 where it reused the same port number. v3.0.1 is in VIPM and fixes that. It might be worth a try.

**Comment by Benni0706** [comment]
The error was that I accidentally had two instances of GitLab Runner running, using the start command and the run command at once. The runner instance I started with the start command does not seem to work. The G-CLI update didn't change this either, but gave me the idea that led to the solution. Thank you for your quick answers!
